accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/ui/WebConsoleConfigTracker.java

```diff
@@ -0,0 +1,88 @@
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = { WebConsoleConfigTracker.class, ConfigurationListener.class })
+public class WebConsoleConfigTracker implements ConfigurationListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WebConsoleConfigTracker.class);
+
+    private static final String CONSOLE_PID = "org.apache.felix.webconsole.internal.servlet.OsgiManager";
+    private static final String CONSOLE_ROOT_PROP = "manager.root";
+    private static final String CONSOLE_ROOT_DEFAULT = "/system/console"; 
+    
+    private static final String CONSOLE_SEC_PROVIDER_PID = "org.apache.sling.extensions.webconsolesecurityprovider.internal.SlingWebConsoleSecurityProvider";
+    private static final String CONSOLE_SEC_PROVIDER_USERS_PROP = "users";
+    private static final String CONSOLE_SEC_PROVIDER_GROUPS_PROP = "groups";
+
+    private static final String[] RELEVANT_PIDS = new String[] {CONSOLE_PID, CONSOLE_SEC_PROVIDER_PID};
+
+    @Reference(policyOption = ReferencePolicyOption.GREEDY)
+    private ConfigurationAdmin configAdmin;
+
+    private String webConsoleRoot;
+
+    private String[] allowedUsers = new String[] {};
+    private String[] allowedGroups = new String[] {};
+
+    @Activate
+    private void updateConfig() {
+        try {
+            Dictionary<String, Object> webconsoleConfig = configAdmin.getConfiguration(CONSOLE_PID).getProperties();
+            if(webconsoleConfig != null) {
+                webConsoleRoot = (String) webconsoleConfig.get(CONSOLE_ROOT_PROP);
+            }
+            webConsoleRoot = StringUtils.defaultIfBlank(webConsoleRoot, CONSOLE_ROOT_DEFAULT);
+
+            Dictionary<String, Object> webconsoleSecProviderConfig = configAdmin.getConfiguration(CONSOLE_SEC_PROVIDER_PID).getProperties();
+            if(webconsoleSecProviderConfig != null) {
+                allowedUsers = PropertiesUtil.toStringArray(webconsoleSecProviderConfig.get(CONSOLE_SEC_PROVIDER_USERS_PROP));
+                allowedGroups = PropertiesUtil.toStringArray(webconsoleSecProviderConfig.get(CONSOLE_SEC_PROVIDER_GROUPS_PROP));
+            }
+            if(LOG.isDebugEnabled()) {
+                LOG.debug("webConsoleRoot: {} allowedUsers: {} allowedGroups: {}", 
+                        webConsoleRoot, 
+                        ArrayUtils.toString(allowedUsers), 
+                        ArrayUtils.toString(allowedGroups));
+            }
+            
+        } catch (IOException e) {
+            LOG.warn("Could not update config: "+e, e);
+        }
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent event) {
+        String pid = event.getPid();
+        if(ArrayUtils.contains(RELEVANT_PIDS, pid)) {
+            updateConfig();
+        }
+    }
+
+    public String getWebConsoleRoot() {
+        return webConsoleRoot;
```

Comment:
I think it is more complicated to return the real path (https://github.com/apache/felix/blob/c2eedf660e7814824c31558f0e3ea8f4082aa720/webconsole/src/main/java/org/apache/felix/webconsole/internal/servlet/OsgiManager.java#L1086). I would rather not expose the link in the Granite UI console. Within the Felix WebConsole it is available as request attribute with name https://github.com/apache/felix/blob/c2eedf660e7814824c31558f0e3ea8f4082aa720/webconsole/src/main/java/org/apache/felix/webconsole/WebConsoleConstants.java#L142.

Reply:
Makes sense, will change to uses the request attribute instead.